DevEdu.API/DevEdu.Business.Tests/CourseServiceTests.cs

```diff
@@ -20,11 +20,136 @@ public class CourseServiceTests
+
+            _courseRepositoryMock.Setup(x => x.DeleteCourse(courseId));
+
+            var sut = new CourseService(_courseRepositoryMock.Object,
+                                        _topicRepositoryMock.Object,
+                                        _taskRepositoryMock.Object,
+                                        _materialRepositoryMock.Object);
+            //When
+            sut.DeleteCourse(courseId);
+
+            //Than
+            _courseRepositoryMock.Verify(x => x.DeleteCourse(courseId), Times.Once());
+        }
+
+        [Test]
+        public void GetCourseById_IntCourseId_ReturnCourseWithGroups()
+        {
+            //Given
+            var courseDto = CourseData.GetCourseDto();
+            var courseId = CourseData.CourseId;
+            _courseRepositoryMock.Setup(x => x.GetCourse(courseId)).Returns(courseDto);
+
+            var sut = new CourseService(_courseRepositoryMock.Object,
+                                        _topicRepositoryMock.Object,
+                                        _taskRepositoryMock.Object,
+                                        _materialRepositoryMock.Object);
+            //When
+            var dto = sut.GetCourse(courseId);
+
+            //Than
+            Assert.AreEqual(courseDto, dto);
+            _courseRepositoryMock.Verify(x => x.GetCourse(1));
+        }
+        [Test]
+        public void GetCourseById_IntCourseId_ReturnCourseWithGroups_Topics_Materials_Tasks()
+        {
+            //Given
+            var courseDto = CourseData.GetCourseDto();
+            var courseId = CourseData.CourseId;
+            _courseRepositoryMock.Setup(x => x.GetCourse(courseId)).Returns(courseDto);
+
+            var sut = new CourseService(_courseRepositoryMock.Object,
+                                        _topicRepositoryMock.Object,
+                                        _taskRepositoryMock.Object,
+                                        _materialRepositoryMock.Object);
+            //When
+            var dto = sut.GetFullCourseInfo(courseId);
+
+            //Than
+            Assert.AreEqual(courseDto, dto);
+            _courseRepositoryMock.Verify(x => x.GetCourse(courseId));
+        }
+
+        [Test]
+        public void GetCourses_WithoutParams_ReturnListCourses()
+        {
+            //Given
+            var courseDto = CourseData.GetListCourses();
+
+            _courseRepositoryMock.Setup(x => x.GetCourses()).Returns(courseDto);
+
+            var sut = new CourseService(_courseRepositoryMock.Object,
+                                        _topicRepositoryMock.Object,
+                                        _taskRepositoryMock.Object,
+                                        _materialRepositoryMock.Object);
+
+            //When
+            var actualCourseDto = sut.GetCourses();
+
+            //Then
+            Assert.AreEqual(courseDto, actualCourseDto);
+            _courseRepositoryMock.Verify(x => x.GetCourses(), Times.Once);
```

Comment:
please add verification for methods:
![image](https://user-images.githubusercontent.com/18329725/127709566-28d4076b-0378-41b9-ad7c-9c6a1eb2e223.png)

Reply:
+